cli/src/cluster_query.rs

```diff
@@ -768,6 +772,32 @@ pub fn process_catchup(
             )
         };
 
+        let average_time_remaining = if total_sleep_interval == 0 {
```

Comment:
nits: I think we can reduce nesting `total_sleep_interval == 0`. That's because the preceding `time_remaining` calculation  doesn't have this condition.

Maybe you just can adjust the following code like this: `if !average_time_remaining.is_normal() || average_time_remaining <= 0.0 {`

Reply:
It's because the preceding `time_remaining` calculation uses only for  showing time remaining (and shows it only if it is positive)
I show (and calculate) not only time, but speed too, and show it when it negative too. I think, if some update will decrease catchup speed, than we will be have some values to compare.

Reply:
i see! this is another good idea behind the code; how about adding a small comment about your reasoning?